src/main/java/org/mockito/internal/verification/argumentmatching/ArgumentMatchingTool.java

```diff
@@ -48,4 +48,28 @@ private static boolean safelyMatches(ArgumentMatcher m, Object arg) {
     private static boolean toStringEquals(ArgumentMatcher m, Object arg) {
         return m.toString().equals(String.valueOf(arg));
     }
+
+    /**
+     * Suspiciously not matching arguments are those that don't match, and the classes have same simple name.
+     */
+    public static Integer[] getNotMatchingArgsWithSameNameIndexes(
```

Comment:
Rather than returning Integers, can we instead return all names that have duplicates? Then we can do the lookup on the name. E.g. return a `Set<String>` here, where the set contains all names that occur at least twice in `arguments`